docs/shared/guides/encapsulated-nx-and-the-wrapper.md

```diff
@@ -0,0 +1,53 @@
+# Encapsulated Nx
+
+`nx init` accepts a flag called `--encapsulated`. When this flag has been passed, Nx manages its own installation separately from any dependencies that are local to your repository. As such, `nx` and any plugins will not be listed in a root level package.json, and also will not be in a root level node_modules folder. In fact, these are not required for an encapsulated Nx repository.
+
+To manage the local Nx installation, and invoke Nx, a total of 4 files are created.
+
+- [.nx/nxw.js](#nx-wrapper)
+- [nx](#nx-and-nxbat)
+- [nx.bat](#nx-and-nxbat)
+- [nx.json](#nx-json)
+
+{% callout type="note" title="Available since Nx v15.7.0" %}
+
+Support for `--encapsulated` was added in Nx v15.7.0. To ensure that it is available, specify the version of nx when running your command so that `npx` doesn't accept an older version that is in the cache. (e.g. `npx nx@15.7.0 init --encapsulated` or `npx nx@latest init --encapsulated`)
+
+{% /callout %}
+
+## Nx Wrapper
+
+When `nx init --encapsulated` is ran, a file is created called `.nx/nxw.js`. This javascript file is responsible for ensuring that your local installation of Nx is up to date with what the repository expects, and then invoking Nx. We refer to this as the "Nx Wrapper".
+
+## Nx and nx bat
+
+`nx` and `nx.bat` perform the same function, with `nx` working under the bash shell and `nx.bat` working for windows users. Each script simply checks that `node` and `npm` are available, and then invokes the [nx wrapper](#nx-wrapper). They should each be committed to your repository, so that users of windows, linux, and macos are able to use Nx.
```

Comment:
```suggestion
`nx` and `nx.bat` perform the same function, with `nx` working under the bash shell and `nx.bat` working for Windows users. Each script simply checks that `node` and `npm` are available, and then invokes the [nx wrapper](#nx-wrapper). They should each be committed to your repository, so that developers on any operating system can use Nx.
```